packages/connect/src/api/firmware/parseFirmwareHeaders.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * parse firmware headers
+ * based on
+ * https://github.com/trezor/trezor-firmware/blob/main/python/src/trezorlib/firmware/vendor.py#L99-L119
+ * https://github.com/trezor/trezor-firmware/blob/main/python/src/trezorlib/firmware/core.py#L63-L88
+ */
+export const parseFirmwareHeaders = (buff: Buffer) => {
+    const vendorHeader = buff.subarray(0, 4).toString('utf8');
+    let trezorImageHeader = '';
+
+    let restbuff: Buffer | undefined;
+    if (vendorHeader === 'TRZV') {
+        const vendorHeaderLength = buff.readUInt32LE(4);
+        trezorImageHeader = buff
+            .subarray(vendorHeaderLength, vendorHeaderLength + 4)
+            .toString('utf8');
+        restbuff = buff.subarray(vendorHeaderLength + 4);
+    } else if (vendorHeader === 'TRZR') {
+        restbuff = buff.subarray(256 + 4);
+        trezorImageHeader = buff.subarray(256 + 4, 256 + 8).toString('utf8');
+    } else {
+        restbuff = buff.subarray(4);
+        trezorImageHeader = buff.subarray(0, 4).toString('utf8');
+    }
```

Comment:
If I understand this correctly, it means roughly:

- based on header magic:
  - in case of `TRZV`, throw away first `vendorHeaderLength` bytes (variable size)
  - in case of `TRZR`, throw away first 256 bytes (constant size)
  - else don't throw away anything
- split the rest into 4 bytes of (nested) magic and the rest
- continue only for `TRZF` header and read the version bytes

If this is right, I believe `TRZR` branch might be incorrect, as a) not 256 but 260 bytes is thrown away (as opposed e.g. to `stripFwHeaders` utility) and b) `restbuff` and `trezorImageHeader` are overlapping.

Reply:
Also, it would be nice to unify it with mentioned `stripFwHeaders` utility, but it's a bit different (in `stripFwHeaders` we need `TRZV` unstripped) so we can leave it for followup.